framework/Steps.py

```diff
@@ -570,6 +570,7 @@ def _localInitializeStep(self,inDictionary):
       else:
         self._outputCollectionLambda.append((lambda x: x[1].addOutput(), outIndex))
         self._outputDictCollectionLambda.append((lambda x: x[1].addOutput(), outIndex))
+    self._registerMetadata(inDictionary)
```

Comment:
This change will break my recent changes in the basic statistics, could you tell me why need to move this?

Reply:
Hm.  I'm guessing it's later in the call, right?

The problem is probably that PostProcessor does not inherit from `BaseClass`, so the methods `provideExpectedMetaKeys` and `addMetaKeys` are missing from the postprocessors.

The short fix is to add those two methods, basically exactly as they are in the Step.

Reply:
We talked in person; the problem was that the sampler was calling `addMetaKeys` in `localGenerateInput` instead of `localInitialize` or `localInputsAndChecks`.